gap/PolygonalComplexes/properties.gi

```diff
@@ -218,6 +218,150 @@ InstallMethod( FaceCounter, "for a twisted polygonal complex",
         return Collected( Compacted( faceDegrees ) );
     end
 );
+
+InstallMethod( ButterflyCounter, "for a simplicial surface",
+    [IsSimplicialSurface],
+    function(surface)
+	local e,temp1,temp2,counter,g,VerticesOfOrthogonalEdge;
+
+        VerticesOfOrthogonalEdge:=function(S,e)
+            local temp;
+            temp:=FacesOfEdge(S,e);
+            temp:=Union(VerticesOfFace(S,temp[1]),VerticesOfFace(S,temp[2]));
+            return Difference(temp, VerticesOfEdge(S,e));
+	end;
+	counter:=[];
+	for e in InnerEdges(surface) do 
+		temp1:=VerticesOfEdge(surface,e);
+		temp1:=SortedList(List(temp1,g->FaceDegreeOfVertex(surface,g)));
+		if not 2 in temp1 then 
```

Comment:
I don't get why you need this case distinction. In my opinion the else-case isn't correct and I think that it is not needed. But maybe I'm missing something.